modules/gapi/test/serialization/serialization.cpp

```diff
@@ -0,0 +1,1292 @@
+{
+
+    os << (uint)cvrects.size();
+    for (auto & element : cvrects)
+    {
+        os << element;
+    }
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, cv::Rect cvrect)
+{
+    os << cvrect.x;
+    os << cvrect.y;
+    os << cvrect.width;
+    os << cvrect.height;
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, std::vector<Data> datas)
+{
+    os << (uint)datas.size();
+    for (const auto& data : datas)
+    {
+        //Data
+        os << data;
+    }
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, Data data)
+{
+    os << data.rc;
+
+    //GMetaArg meta;
+    switch (data.rc.shape)
+    {
+    case cv::GShape::GMAT:
+    {
+        GMatDesc mat_desc = util::get<GMatDesc>(data.meta);
+        os << mat_desc.depth;
+        os << mat_desc.chan;
+        os << mat_desc.size.width;
+        os << mat_desc.size.height;
+
+        int bool_val;
+        bool_val = mat_desc.planar ? 1 : 0;
+        os << bool_val;
+
+        uint dims_size = (uint)mat_desc.dims.size();
+        os << dims_size;
+        for (uint j = 0; j < dims_size; j++)
+        {
+            os << (uint)mat_desc.dims.data()[j];
+        }
+        break;
+    }
+    case cv::GShape::GSCALAR:
+    {
+        //std::cout << "dumpGSerializedDatas  GSCALAR " << std::endl;
+        //GScalarDesc scalar_desc = util::get<GScalarDesc>(data.meta);
+        break;
+    }
+    case cv::GShape::GARRAY:
+    {   //std::cout << "dumpGSerializedDatas  GARRAY " << std::endl;
+        //GArrayDesc array_desc = util::get<GArrayDesc>(data.meta);
+        break;
+    }
+    case cv::GShape::GOPAQUE:
+    {
+        //std::cout << "dumpGSerializedDatas  GOPAQUE " << std::endl;
+        //GOpaqueDesc opaque_desc = util::get<GOpaqueDesc>(data.meta);
+        break;
+    }
+    default:
+        //std::cout << "dumpGSerializedDatas  unsupported" << std::endl;
+        break;
+    }
+
+    return os;
+}
+
+SerializationStream& operator << (SerializationStream& os, Op op)
+{
+    //Kernel k
+    os << op.k;
+
+    //std::vector<int>   kind;
+    os << op.kind;
+
+    //std::vector<int>   opaque_kind;
+    os << op.opaque_kind;
+
+    //std::vector<RcDesc> outs;
+    os << op.outs;
+
+    //std::vector<RcDesc> ins;
+    os << op.ins;
+
+    //opaque args
+    //std::vector<int> opaque_ints;
+    os << op.opaque_ints;
+
+    //std::vector<double> opaque_doubles;
+    os << op.opaque_doubles;
+
+    //std::vector<cv::Size> opaque_cvsizes;
+    os << op.opaque_cvsizes;
+
+    //std::vector<bool> opaque_bools;
+    os << op.opaque_bools;
+
+    //std::vector<cv::Scalar> opaque_cvscalars;
+    os << op.opaque_cvscalars;
+
+    //std::vector<cv::Point> opaque_cvpoints;
+    os << op.opaque_cvpoints;
+
+    //std::vector<cv::Mat> opaque_cvmats;
+    os << op.opaque_cvmats;
+
+    //std::vector<cv::Rect> opaque_cvrects;
+    os << op.opaque_cvrects;
```

Comment:
Too much stuff here. This can be simplified (I mean, the whole `Op` serialization). But leave it as is now

Reply:
OK